### ISSUE TITLE
Broken or Flaky test: ExternalCompaction_2_IT and ExternalCompaction_3_IT

### ISSUE THREAD

**Issue by markjens** [apache/accumulo#2375]
**Test name(s)**
 - ExternalCompaction_2_IT
 - ExternalCompaction_3_IT

**Describe the failure observed**

ExternalCompaction_2_IT and ExternalCompaction_3_IT hang and timeout on Linux ARM64.
These two tests always timeout on my test system, no matter how much time I give them (`-Dtimeout.factor=...`)
Thread dumps of all processes could be found at https://gist.github.com/markjens/326b681c3f8a9c1b6400f55847fe7716. I've taken several dumps with ~5secs interval between them.
I do not see anything suspicious in the stack traces. That's why I need your help to review them and hopefully find what the problem is!

**Testing Environment:**
 - Version of this project: 2.1.0-SNAPSHOT
 - First commit known to fail (or current commit): current `main` branch
 - OS: Ubuntu 20.04.3 
 - Java: 11.0.11
 - Maven: 3.8.2
 - Other environment details: Linux ARM64

**What have you tried already?**
- mvn clean install -DskipTests
- mvn verify (ExternalCompaction_2_IT and ExternalCompaction_3_IT timed out)
- mvn clean verify -Dit.test=ExternalCompaction_3_IT -Dtimeout.factor=3 -o -Pfast-build -N 

<details><summary>Tests output</summary>
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.accumulo.test.compaction.ExternalCompaction_3_IT
[ERROR] Tests run: 4, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 3,600.677 s <<< FAILURE! - in org.apache.accumulo.test.compaction.ExternalCompaction_3_IT
[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testMergeCancelsExternalCompaction  Time elapsed: 1800.016 s  <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 1800 seconds
	at java.base@17.0.1/java.lang.Thread.sleep(Native Method)
	at app//org.apache.accumulo.fate.util.UtilWaitThread.sleep(UtilWaitThread.java:33)
	at app//org.apache.accumulo.test.compaction.ExternalCompactionTestUtils.confirmCompactionCompleted(ExternalCompactionTestUtils.java:329)
	at app//org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testMergeCancelsExternalCompaction(ExternalCompaction_3_IT.java:113)
	at java.base@17.0.1/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@17.0.1/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base@17.0.1/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@17.0.1/java.lang.reflect.Method.invoke(Method.java:568)
	at app//org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at app//org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at app//org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at app//org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at app//org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at app//org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)

[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testMergeCancelsExternalCompaction  Time elapsed: 1800.017 s  <<< ERROR!
java.lang.Exception: Appears to be stuck in thread Time-limited test-SendThread(localhost:35839)
	at java.base@17.0.1/sun.nio.ch.EPoll.wait(Native Method)
	at java.base@17.0.1/sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:118)
	at java.base@17.0.1/sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:129)
	at java.base@17.0.1/sun.nio.ch.SelectorImpl.select(SelectorImpl.java:141)
	at app//org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:347)
	at app//org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1223)

[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testCoordinatorRestartsDuringCompaction  Time elapsed: 1800.001 s  <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 1800 seconds
	at java.base@17.0.1/java.lang.Thread.sleep(Native Method)
	at app//org.apache.accumulo.fate.util.UtilWaitThread.sleep(UtilWaitThread.java:33)
	at app//org.apache.accumulo.test.compaction.ExternalCompactionTestUtils.startCoordinator(ExternalCompactionTestUtils.java:243)
	at app//org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testCoordinatorRestartsDuringCompaction(ExternalCompaction_3_IT.java:141)
	at java.base@17.0.1/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@17.0.1/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base@17.0.1/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@17.0.1/java.lang.reflect.Method.invoke(Method.java:568)
	at app//org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at app//org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at app//org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at app//org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at app//org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at app//org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at app//org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base@17.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base@17.0.1/java.lang.Thread.run(Thread.java:833)

[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testCoordinatorRestartsDuringCompaction  Time elapsed: 1800.002 s  <<< ERROR!
java.lang.Exception: Appears to be stuck in thread Time-limited test-SendThread(localhost:41655)
	at java.base@17.0.1/sun.nio.ch.EPoll.wait(Native Method)
	at java.base@17.0.1/sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:118)
	at java.base@17.0.1/sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:129)
	at java.base@17.0.1/sun.nio.ch.SelectorImpl.select(SelectorImpl.java:141)
	at app//org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:347)
	at app//org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1223)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testCoordinatorRestartsDuringCompaction
[ERROR]   Run 1: ExternalCompaction_3_IT.testCoordinatorRestartsDuringCompaction:141 » TestTimedOut
[ERROR]   Run 2: ExternalCompaction_3_IT.testCoordinatorRestartsDuringCompaction »  Appears to ...
[INFO] 
[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_3_IT.testMergeCancelsExternalCompaction
[ERROR]   Run 1: ExternalCompaction_3_IT.testMergeCancelsExternalCompaction:113 » TestTimedOut ...
[ERROR]   Run 2: ExternalCompaction_3_IT.testMergeCancelsExternalCompaction »  Appears to be st...
[INFO] 
[INFO] 
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0
</details>

**Additional context**

Please let me know if you need more information!

**Comment by EdColeman** [comment]
Running ExternalCompaction_2_IT, the test testSplitCancelsExternalCompaction seems to be hitting the bug described in issue #2362.  With the test running it looks like from a quick look:

The Manager and Tservers are waiting for something to happen.
The UncaughtExcption handler error message is in the the ExternalDoNothingCompactor log
The ExternalDoNothingCompactor keeps reporting `compactor.Compactor] ERROR: Waiting on compaction thread to finish, but no RUNNING compaction`

**Comment by EdColeman** [comment]
For org.apache.accumulo.test.compaction.ExternalCompaction_2_IT three tests failed with timeouts:
testSplitCancelsExternalCompaction - has uncaught exception
testDeleteTableCancelsUserExternalCompaction - has uncaught exception
testUserCompactionCancellation

testUserCompactionCancellation failed because it could not start the coordinator service

# From CompactionCoordinator_##.out
<details>
  <summary>Click to expand!</summary>
  org.apache.thrift.transport.TTransportException: Unable to create server on addresses: [0.0.0.0:9132]
        at org.apache.accumulo.server.rpc.TServerUtils.startTServer(TServerUtils.java:669) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.rpc.TServerUtils.startTServer(TServerUtils.java:601) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.rpc.TServerUtils.startServer(TServerUtils.java:185) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.coordinator.CompactionCoordinator.startCoordinatorClientService(CompactionCoordinator.java:231) ~[accumulo-compaction-coordinator-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.coordinator.CompactionCoordinator.run(CompactionCoordinator.java:246) ~[accumulo-compaction-coordinator-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:207) ~[opentelemetry-context-1.7.1.jar:1.7.1]
        at java.lang.Thread.run(Thread.java:829) [?:?]
2021-12-06T13:14:20,348 [start.Main] ERROR: Thread 'org.apache.accumulo.coordinator.CompactionCoordinator' died.
java.lang.RuntimeException: Failed to start the coordinator service
        at org.apache.accumulo.coordinator.CompactionCoordinator.run(CompactionCoordinator.java:248) ~[accumulo-compaction-coordinator-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:207) ~[opentelemetry-context-1.7.1.jar:1.7.1]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.net.UnknownHostException: Unable to find a listen port
        at org.apache.accumulo.server.rpc.TServerUtils.startServer(TServerUtils.java:222) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.coordinator.CompactionCoordinator.startCoordinatorClientService(CompactionCoordinator.java:231) ~[accumulo-compaction-coordinator-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.coordinator.CompactionCoordinator.run(CompactionCoordinator.java:246) ~[accumulo-compaction-coordinator-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        ... 2 more

</details>

For ExternalCompaction_3_IT_testUserCompactionCancellation - uncaught exception is in the log

**Comment by markjens** [comment]
Thank you for looking into this, @EdColeman !

Did you see `TTransportException: Unable to create server on addresses` in my logs or in your local run of these tests ?
Because I see no such error at https://gist.githubusercontent.com/markjens/326b681c3f8a9c1b6400f55847fe7716/raw/59a33655e108fe55708ceb0d9f5c81309d056a35/CompactionCoordinator_1415671774.out

I see this one:
```
2021-12-03T10:42:21,972 [zookeeper.ZooCache] WARN : Saw (possibly) transient exception communicating with ZooKeeper, will retry
org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss for /accumulo/77a5af20-706e-4a5b-9adf-e188329b760f/root_tablet
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:102) ~[zookeeper-3.5.9.jar:3.5.9]
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:54) ~[zookeeper-3.5.9.jar:3.5.9]
	at org.apache.zookeeper.ZooKeeper.exists(ZooKeeper.java:2021) ~[zookeeper-3.5.9.jar:3.5.9]
	at org.apache.accumulo.fate.zookeeper.ZooCache$2.run(ZooCache.java:415) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
```
but since it is a WARN I assumed it is not the problem.

**Comment by milleruntime** [comment]
I am still seeing ExternalCompaction_2_IT fail on the latest changes, even with the fix for #2362. Although, I am not sure it is the same problem.
<pre>
2022-02-01T07:44:41,450 [compaction.FileCompactor] ERROR: File file:/home/mike/workspace/accumulo/test/target/mini-tests/org.apache.accumulo.test.compaction.ExternalCompacti
on_2_IT_testDeleteTableCancelsExternalCompaction/accumulo/tables/+r/root_tablet/00000_00000.rf does not exist
java.io.FileNotFoundException: File file:/home/mike/workspace/accumulo/test/target/mini-tests/org.apache.accumulo.test.compaction.ExternalCompaction_2_IT_testDeleteTableCancelsExternalCompaction/accumulo/tables/+r/root_tablet/00000_00000.rf does not exist
        at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:668) ~[hadoop-client-api-3.3.0.jar:?]
        at org.apache.hadoop.fs.RawLocalFileSystem.getFileLinkStatusInternal(RawLocalFileSystem.java:989) ~[hadoop-client-api-3.3.0.jar:?]
        at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:658) ~[hadoop-client-api-3.3.0.jar:?]
        at org.apache.hadoop.fs.RawLocalFileSystem.open(RawLocalFileSystem.java:214) ~[hadoop-client-api-3.3.0.jar:?]
        at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:945) ~[hadoop-client-api-3.3.0.jar:?]
        at org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile$CachableBuilder.lambda$fsPath$0(CachableBlockFile.java:87) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile$Reader.getBCFile(CachableBlockFile.java:163) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile$Reader.getMetaBlock(CachableBlockFile.java:392) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.rfile.RFile$Reader.<init>(RFile.java:1164) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.rfile.RFile$Reader.<init>(RFile.java:1256) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.rfile.RFileOperations.getReader(RFileOperations.java:55) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.rfile.RFileOperations.openReader(RFileOperations.java:70) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.DispatchingFileFactory.openReader(DispatchingFileFactory.java:85) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.file.FileOperations$ReaderBuilder.build(FileOperations.java:447) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.compaction.FileCompactor.openMapDataFiles(FileCompactor.java:298) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.compaction.FileCompactor.compactLocalityGroup(FileCompactor.java:344) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.compaction.FileCompactor.call(FileCompactor.java:224) [accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.tserver.tablet.CompactableUtils.compact(CompactableUtils.java:560) [accumulo-tserver-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.tserver.tablet.CompactableImpl.compact(CompactableImpl.java:1260) [accumulo-tserver-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.tserver.compactions.InternalCompactionExecutor$InternalJob.run(InternalCompactionExecutor.java:96) [accumulo-tserver-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) [accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) [accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:829) [?:?]
</pre>
